xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -4218,11 +4216,11 @@ void CDVDPlayer::GetAudioStreamInfo(int index, SPlayerAudioStreamInfo &info)
   }
 }
 
-int CDVDPlayer::AddSubtitleFile(const std::string& filename, const std::string& subfilename, CDemuxStream::EFlags flags)
+int CDVDPlayer::AddSubtitleFile(const std::string& filename, const std::string& subfilename)
```

Comment:
isn't GetNrOfSubtitleStreams() a method of demuxer?
as you pointed out, m_SelectionStreams should be accessed by dvdplayer thread only too. Yes, I think you should add a message. For setting the subtitle we have a message as well.